### PR TITLE
Feat: implement API authorization (GHORG-24)

### PR DESCRIPTION
**Pull request by nguyentoanit**

### CODE REVIEW

**Comment by moltar** [comment]
Task linked: [GHORG-24 SP API authorization](https://app.clickup.com/t/2250003/GHORG-24)

File: src/types/api-clients/selling-partner-api-model-properties.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { SellingPartnerAPICredentials } from './selling-partner-api-credentials'
+
+export interface SellingPartnerApiModelProperties {
+  service?: string
```
Comment (moltar) [inline src/types/api-clients/selling-partner-api-model-properties.ts:4]: We don't need `service` it is always static.

**Comment by nguyentoanit** [comment]
@moltar : Thanks for your suggestions.
I used `awsRegion` from `@scaleleap/amazon-marketplaces` package for testing and updated Basic Usage docs too.

File: test/integration/sellers-api-client.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            sessionToken: Credentials!.SessionToken || '',
+          },
+        },
+      })
+
+      /**
+       * TODO: response does not match types.
+       */
+
+      const response = await client.getMarketplaceParticipations()
```
Comment (moltar) [inline test/integration/sellers-api-client.test.ts:64]: Here, `response` seems to be an `AxiosResponse`, but the API types are returning `response.data`. There is a type mismatch somewhere.

**Review by moltar — CHANGES_REQUESTED** [review]

**Comment by moltar** [comment]
@nguyentoanit please take a look at my commit and one comment.

File: test/integration/sellers-api-client.test.ts (hunk 2 above)
Comment (nguyentoanit) [reply test/integration/sellers-api-client.test.ts]: @moltar : 
It seem to be not able to override response type when we unwrap `AxiosResponse` in axios interceptors.

In perhaps, we need to override `AxiosResponse`:
```
import axios from 'axios'

declare module 'axios' {
  export interface AxiosResponse<T = any> extends Promise<T> {}
}
```

But I don't think that is good idea. Therefore, i removed unwrapped `AxiosResponse` in axios interceptors.

Ref: https://github.com/axios/axios/issues/1510#issuecomment-448201698

File: test/integration/sellers-api-client.test.ts [hunk 3]
```diff
@@ -57,17 +57,11 @@ describe(`${SellersApiClient.name}`, () => {
         },
       })
 
-      /**
-       * TODO: response does not match types.
-       */
+      const marketplaceParticipations = await client
```
Comment (moltar) [inline test/integration/sellers-api-client.test.ts:60]: FYI can also be written as:

```ts
const { data: marketplaceParticipations } = await client.getMarketplaceParticipations()
```

**Review by moltar — CHANGES_REQUESTED** [review]
Please look at the other TODOs as well. Notably: ` // TODO: fix these warnings`. I just stubbed all of it to work, but please complete it to avoid warnings.

**Comment by moltar** [comment]
Coverage tests are failing, because they do not have environment variables.

We should probably separate integration tests from the rest of the tests, and run them separately. But we can do that in a separate PR. So this is ok to merge.

**Review by moltar — APPROVED** [review]

**Pull request merged**